### PR TITLE
window layout

### PR DESCRIPTION
**Pull request by madflow**
* introduce window layout presets
* not sure how this can be tested :S

### CODE REVIEW

**Comment by jubnzv** [comment]
Hi @madflow,
Thanks for the PR! It seems completely reasonable for me.

The only problem here is that it changes the API of the library breaking the backward compatibility. I have no idea who uses this library, but in general such changes assume bumping the major version of the package (according to [semver](https://semver.org)) and should be done only when it is really needed.

Therefore, I have the following suggestion: let's make a `SetLayout` method of `Window`, and assume, that the default layout set in `NewWindow` is an empty string. Following this pattern, we can add more methods to the internal structs in the future, without breaking the backward compatibility, only adding new interfaces.

What do you think?

File: configuration.go [hunk 1]
```diff
@@ -109,14 +111,25 @@ func (c *Configuration) Apply() error {
...
+			// Select layout if defined
+			// https://www.man7.org/linux/man-pages/man1/tmux.1.html#WINDOWS_AND_PANES
+			if len(w.Layout) != 0 {
+				args = []string{"select-layout", "-t", winId, w.Layout}
+				_, _, err_exec := RunCmd(args)
+				if err_exec != nil {
+					return err_exec
+				}
+			}
+
```
Comment (jubnzv) [inline configuration.go]: nit: please remove an empty line

**Comment by madflow** [comment]
I understand (I think) ;)

So the idea is:

* Remove `layout` from the  `NewWindow`  creator method.
* Replace `Layout:         layout,` with `Layout:         "",`
* Implement

```go
// window.go
func (w *Window) SetLayout(layout string) {
	w.Layout = layout
}
```
* Keep the configuration section intact

?

Or should `SetLayout` signal tmux directly?

**Comment by jubnzv** [comment]
Yes, that's right, I suggest to add a setter method and change the constructor.

**Comment by madflow** [comment]
I have changed to PR with the a new setter method. I wonder if it is weird for the user, that the Layout will always be empty, when the window is "listed"? Also you can only apply it, in the configuration phase. It will be useful in any case.

File: window.go [hunk 2]
```diff
@@ -37,3 +47,11 @@ func (w *Window) ListPanes() ([]Pane, error) {
 func (w *Window) AddPane(pane Pane) {
 	w.Panes = append(w.Panes, pane)
 }
+
+// Sets the window layout. The possible value can be one of the contants:
+// LayoutEvenVertical, LayoutEvenHorizontal, LayoutMainVertical,  LayoutMainHorizontal, LayoutTiled
+// See: https://www.man7.org/linux/man-pages/man1/tmux.1.html#WINDOWS_AND_PANES
+
```
Comment (jubnzv) [inline window.go]: Please remove a new line to make a docstring comment from it

**Comment by jubnzv** [comment]
Thanks! Just one small nit.

> I wonder if it is weird for the user, that the Layout will always be empty, when the window is "listed"

I think that's okay for Golang. In a language with the `Option` type, there would be the `None` value.

File: window.go (hunk 2 above)
Comment (madflow) [reply window.go]: fixed

**Pull request merged**